src/utils/transaction.ts

```diff
@@ -56,9 +58,9 @@ export function getTxSenderAddress(
     }
 
     const rawTx = '0x' + toHexString(tx.serialize())
-    const { address, isValid } = getSenderAddress(rawTx)
+    const { address, isValid, gasValid } = getSenderAddress(rawTx)
     if (logFlags.dapp_verbose) console.log('Sender address retrieved from signed txn', address)
-    const res = { address: Address.fromString(address), isValid }
+    const res = { address: Address.fromString(address), isValid, gasValid }
     if (txid != null) {
       simpleTTL++
       if (simpleTTL > cacheMaxSize) {
```

Comment:
I think there is a bug in cache cleanup logic in this function. why it is set to cacheMaxSize when cache is cleared? it will be cleared on each tx after.